tuplex/codegen/src/FunctionRegistry.cc

```diff
@@ -1628,6 +1631,116 @@ namespace tuplex {
             return SerializableValue();
         }
 
+        SerializableValue FunctionRegistry::createIteratorRelatedSymbolCall(tuplex::codegen::LambdaFunctionBuilder &lfb,
+                                                                            llvm::IRBuilder<> &builder,
+                                                                            const std::string &symbol,
+                                                                            const python::Type &argsType,
+                                                                            const python::Type &retType,
+                                                                            const std::vector<tuplex::codegen::SerializableValue> &args,
+                                                                            const std::shared_ptr<IteratorInfo> &iteratorInfo) {
+            if(symbol == "iter") {
+                return createIterCall(lfb, builder, argsType, retType, args);
+            }
+
+            if(symbol == "reversed") {
+                return createReversedCall(lfb, builder, argsType, retType, args);
+            }
+
+            if(symbol == "zip") {
+                return createZipCall(lfb, builder, argsType, retType, args, iteratorInfo);
+            }
+
+            if(symbol == "enumerate") {
+                return createEnumerateCall(lfb, builder, argsType, retType, args, iteratorInfo);
+            }
+
+            if(symbol == "next") {
+                return createNextCall(lfb, builder, argsType, retType, args, iteratorInfo);
+            }
+
+            Logger::instance().defaultLogger().error("unsupported symbol encountered");
```

Comment:
maybe add in error message what symbol, i.e.

`"unsupported symbol '" + symbol + "' encountered in createIteratorRelatedSymbolCall"`

Reply:
Updated.